### PR TITLE
Bug: Dangling References to Deleted Projects

### PR DESCRIPTION
**Pull request by bryce-fitzsimons**
### Features and Changes

- Closes **GB-169**

- Strong warnings on SDK Connections pages (list page, connection page, edit modal)
- Strong warning when deleting project
- Avoid refreshing the SDK payload cache when project is orphaned (to buy some time until the user can fix it)

### Screenshots

image
image
image
Screen Shot 2023-04-24 at 10 44 17 PM

### CODE REVIEW

File: packages/back-end/src/models/SdkPayloadModel.ts [hunk 1]
```diff
@@ -83,6 +85,14 @@ export async function updateSDKPayload({
     experiments: experimentsDefinitions,
   };
 
+  const projectDoc = project
+    ? await findProjectById(project, organization)
+    : null;
+  if (project && !projectDoc) {
+    logger.error(`Cannot update SDK payload: project ${project} not found`);
+    return;
+  }
```
Comment (jdorn) [inline packages/back-end/src/models/SdkPayloadModel.ts:94]: Is there a way we can catch this earlier?  If someone deletes a project that's used by 20 features, it will call `onFeatureUpdated` 20 times.  Each of those is going to call this function with the now-deleted project.  And that means 20 additional Mongo queries checking if the project exists and 20 error logs, even though it's expected behavior and not an error.

Plus, every feature update in a valid project will also run an additional Mongo query now.

Comment (bryce-fitzsimons) [reply packages/back-end/src/models/SdkPayloadModel.ts]: Thanks for the catch, updated.

File: packages/front-end/components/Layout/Layout.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ const navlinks: SidebarLinkProps[] = [
     autoClose: true,
     subLinks: [
       {
-        name: "All Features",
+        name: "Features",
```
Comment (jdorn) [inline packages/front-end/components/Layout/Layout.tsx:36]: Revert this change for now.  I have a whole left nav restructure in #1145

**Review by jdorn — APPROVED** [review]

**Review by jdorn — APPROVED** [review]
